editor/effects.js

```diff
@@ -273,4 +276,8 @@ export default {
 
 		return effects;
 	},
+	RESET_BLOCKS( action, store ) {
+		const blocks = [ createBlock( 'core/post-title' ) ].concat( action.blocks );
+		store.dispatch( { type: 'ACTUALLY_RESET_BLOCKS', blocks } );
```

Comment:
If we need to better distinguish between effects triggered by `SET_INITIAL_POST` and what is considered in resetting blocks, I think we should either come up with better names (avoid "ACTUALLY_" prefix) or a different pattern for manipulating how post title is inserted into the block set. A standalone middleware which modifies `action.blocks` before dispatch occurs is one option.